xclim/core/utils.py

```diff
@@ -162,3 +164,208 @@ def sfcwind_2_uas_vas(wind: xr.DataArray = None, windfromdir: xr.DataArray = Non
+    the dew point temperature, the specific humidity and the air pressure.
+
+    **For the "dewpoint" method** : With :math:`L` the Enthalpy of vaporization of water
+    and :math:`R_w` the gas constant for water vapor, the relative humidity is computed as:
+
+    .. math::
+
+        RH = e^{\\frac{-L (T - T_d)}{R_wTT_d}}
+
+    Formula taken from [Lawrence_2005]_.
+
+    **Other methods**: With :math:`w`, :math:`w_{sat}`, :math:`e_{sat}` the mixing ratio,
+    the saturation mixing ratio and the saturation partial pressure of water vapor, relative humidity is computed as:
+
+        ... math::
+
+            RH = 100\\frac{w}{w_{sat}}
+            w = \\frac{q}{1-q}
+            w_{sat} = 0.622\\frac{e_{sat}}{P - e_{sat}}
+
+    The methods differ by how :math:`e_{sat}` is computed. In all cases, :math:`log(e_{sat})`
+    is an empirically fitted function (usually a polynomial) where coefficients can be different
+    when ice is taken as refernce instead of water. Available methods are:
+
+    - "goffgratch46" or "GG46", based on [GoffGratch_1946]_, values taken from [Voemel]_.
+    - "sonntag90" or "SO90", taken from [Sonntag_1990]_.
+    - "tetens30" or "TE30", based on [Tetens_1930], values taken from [Voemel]_.
+    - "wmo08" or "WMO08", taken from [WMO_2008]_.
+
+
+    References
+    ----------
+    .. [GoffGratch_1946] Goff, J. A., and S. Gratch (1946) Low-pressure properties of water from -160 to 212 °F, in Transactions of the American Society of Heating and Ventilating Engineers, pp 95-122, presented at the 52nd annual meeting of the American Society of Heating and Ventilating Engineers, New York, 1946.
+    .. [Lawrence_2005] Lawrence, M.G. (2005). The Relationship between Relative Humidity and the Dewpoint Temperature in Moist Air: A Simple Conversion and Applications. Bull. Amer. Meteor. Soc., 86, 225–234, https://doi.org/10.1175/BAMS-86-2-225
+    .. [Sonntag_1990] Sonntag, D. (1990). Important new values of the physical constants of 1986, vapour pressure formulations based on the ITS-90, and psychrometer formulae. Zeitschrift für Meteorologie, 40(5), 340-344.
+    .. [Tetens_1930] Tetens, O. 1930. Über einige meteorologische Begriffe. Z. Geophys 6: 207-309.
+    .. [Voemel] http://cires1.colorado.edu/~voemel/vp.html
+    .. [WMO_2008] World Meteorological Organization. (2008). Guide to meteorological instruments and methods of observation. Geneva, Switzerland: World Meteorological Organization. https://www.weather.gov/media/epz/mesonet/CWOP-WMO8.pdf
+    """
+    if dtas is not None:
+        dtas = convert_units_to(dtas, "degK")
+        tas = convert_units_to(tas, "degK")
+        L = 2.501e6
+        Rw = (461.5,)
+        rh = 100 * np.exp(-L * (tas - dtas) / (Rw * tas * dtas))
+        desc = (
+            "Relative humidity computed from temperature and "
+            f"dew point temperature with L = {L} and Rw = {Rw}"
+        )
+        method = "dewpoint"
```

Comment:
Wait, why is setting `dtas` changing the value of `method` ? `method` is an input, no?

Reply:
It's because "method" is irrelevant when computing RH from the dew point temp. Thus, I override the method so it can be formatted correctly further down. It's explained in the doc, but we could also raise a ValueError on wrong argument inputs.

Reply:
It does, which is why its now the case ;)